### PR TITLE
[Docs] Only display H2 headings in ToC by default

### LINKED ISSUE #5438: [Docs] Add ToC level filter so not all levels are displayed

### PR DESCRIPTION
**Pull request by rmoff**
### Linked Issue

Closes #5438

---

## Change Description

tl;dr: This is a minimal-impact change that changes in the background how ToC are generated on docs pages. You can see the impact of it here: https://treeverse-lakefs-preview-pr-6046.surge.sh/. In most cases you won't even see a difference. Where you will notice a difference is where pages are now more readable and not drowned in excessive-length ToC, for example: 

CleanShot_2023-06-09_at_15 41 11@2x

For details of what's changed and why, please read on…

### Background

Currently docs pages have a Table of Contents added to them by the inclusion of the following at the point in the markdown at which we want the ToC rendered: 

```
{% include toc.html %}
```

This, in turn, invokes `{:toc}` which is [used by Kramdown](https://kramdown.gettalong.org/converter/html.html#toc) (the Markdown processor that Jekyll uses) to insert a table of contents from the H1-H6 headings present in the markdown. 

There is no easy way to limit the level of headings included, other than the manual use of `{: .no_toc }`. To exclude a heading from being included in the ToC requires the addition of `{: .no_toc }` after the heading. This ends up being added after every page title (`H1`), as well on an ad-hoc basis where the number of subheadings makes a ToC unwieldy: 

image
      
### New Feature

Kramdown supports the `toc_levels` configuration option. This PR adds that as a default to the existing `toc.html` to fix ToCs to level H2 only. Since some pages will want to use H3 also a new file has been added which can be used instead

```
{% include toc_2-3.html %}
```

This PR also removes the now-redundant `{: .no_toc }` which makes files easier to grok and maintain by humans. 

Included in the PR is necessary fixing of some docs that had incorrect heading structures (e.g. no H1, or duplicated H1). 

### Testing Details

Changes checked manually on local build, as well as [the PR preview](https://treeverse-lakefs-preview-pr-6046.surge.sh/) 

Also ran a diff on the generated HTML before/after and checked all ToCs are sensibly rendered (i.e. catch any where we _should_ have >h2 included)

CleanShot_2023-06-09_at_17 17 31

### Breaking Change?

Only for docs authors, who now no longer need to include `{: .no_toc }` all the time. 

## Additional info

The other option that I looked at was https://github.com/allejo/jekyll-toc but discounted for two reasons: 

1. It's an additional dependency to understand and maintain
2. By default it's only available at the template level, not per-page, and so would make customising ToC levels impossible

### CODE REVIEW

File: docs/integrations/duckdb.md [hunk 1]
```diff
@@ -9,14 +9,13 @@ has_children: false
...
+[DuckDB](https://duckdb.org/){:target="_blank"} is an in-process SQL OLAP database management system.
+
 {% include toc.html %}
 
 ## Configuration
 
-Querying data in lakeFS from DuckDB is similar to querying data in S3 from DuckDB. It is done using the [httpfs extension](https://duckdb.org/docs/extensions/httpfs.html){:target="_blank"}.
-
-### Configure the lakeFS S3 Gateway endpoint
+Querying data in lakeFS from DuckDB is similar to querying data in S3 from DuckDB. It is done using the [httpfs extension](https://duckdb.org/docs/extensions/httpfs.html){:target="_blank"} connecting to the S3 Gateway that lakeFS provides.
```
Comment (ozkatz) [inline docs/integrations/duckdb.md:18]: Perhaps we should link to an explanation on the s3 gateway. Could be in a different PR.

File: docs/integrations/python.md [hunk 2]
```diff
@@ -11,11 +11,12 @@ redirect_from:
   - /integrations/boto.html
 ---
 
-{% include toc.html %}
+# Use Python to interact with your objects on lakeFS
 
-## Boto vs. lakeFS SDK
+{% include toc_2-3.html %}
+
+There are two primary ways to work with lakeFS from Python:
```
Comment (ozkatz) [inline docs/integrations/python.md:18]: I feel boto should be a different integration, we should probably default to the native python SDK. (again we can discuss outside this PR)

**Review by ozkatz — APPROVED** [review]
Definitely an improvement!

File: docs/integrations/python.md (hunk 2 above)
Comment (rmoff) [reply docs/integrations/python.md:18]: Tracking: https://github.com/treeverse/lakeFS/issues/6056

File: docs/integrations/duckdb.md (hunk 1 above)
Comment (rmoff) [reply docs/integrations/duckdb.md:18]: Done
